libs/@hashintel/petrinaut/src/core/README.md

```diff
@@ -0,0 +1,108 @@
+# SDCPN prototype
+
+This library contains core functionalities for simulating Stochastic Deterministic Colored Petri Nets (SDCPNs). It provides data structures and algorithms to define, manipulate, and execute SDCPN models.
+
+```ts
+import { createSDCPN, createSimulation } from "@hashintel/petrinaut";
+
+const sdcpn = createSDCPN({
+  id: "example-sdcpn",
+  name: "Example SDCPN Model",
+  places: [
+    {
+      id: "p-space",
+      name: "Space",
+      dimensions: [
+        { name: "x", type: "real" },
+        { name: "y", type: "real" },
+        { name: "direction", type: "real" },
+        { name: "velocity", type: "real" },
+      ],
+      differentialEquation: `export default Dynamics((tokens) => {
+        const mu = 400000.0; // Gravitational parameter
+
+        // Return derivatives for each dimension of each token
+        return tokens.map(([x, y, angle, velocity]) => {
+          const r = Math.hypot(x, y); // Distance to Earth center
+    
+          // Gravitational acceleration vector (points toward origin)
+          const ax = (-mu * x) / (r * r * r);
+          const ay = (-mu * y) / (r * r * r);
+          
+          // Project that acceleration into satellite’s velocity frame
+          const ddirection =
+            (-ax * Math.sin(direction) + ay * Math.cos(direction)) / velocity;
+          const dvelocity = ax * Math.cos(direction) + ay * Math.sin(direction);
+          
+          // Position derivative (from velocity)
+          const dx = velocity * Math.cos(direction);
+          const dy = velocity * Math.sin(direction);
+
+          return [dx, dy, ddirection, dvelocity];
+        });
+      })`,
+    },
+    {
+      id: "p-cemetery",
+      name: "Cemetery",
+      dimensions: [],
+      // No dynamics needed for cemetery
+      // nor dimensions
+    },
+  ],
+  transitions: [
+    {
+      id: "t-collision",
+      name: "Collision",
+      inputArcs: [{ placeId: "p-space", weight: 2 }],
+      outputArcs: [{ placeId: "p2", weight: 2 }],
+      lambdaCode: `export default Lambda(([s1, s2]) => {
```

Comment:
Can we default to using descriptive variable names rather than single letters please?

In this case, I guess `satellite1` and `satellite2`?